components/logins/src/db.rs

```diff
@@ -577,12 +609,43 @@ impl LoginDb {
                 Ok(_) => log::info!("Imported {} (new GUID {}) successfully.", old_guid, guid),
                 Err(e) => {
                     log::warn!("Could not import {} ({}).", old_guid, e);
-                    num_failed += 1;
+                    insert_errors.push(format!("Could not import {} ({}).", old_guid, e));
+                    num_failed_insert += 1;
                 }
             };
         }
         tx.commit()?;
-        Ok(num_failed)
+        // TODO: handle potential negative durations and counts
+        let fixup_phase_duration = fixup_phase_end - import_start;
+        let num_post_fixup = import_start_total_logins - num_failed_fixup;
```

Comment:
Since it *should* be impossible for these counts to go negative, I think I'm OK with leaving this as-is, but would also support `.checked_sub().unwrap()` or `.saturating_sub()` if you prefer.